runtime/parachains/src/paras.rs

```diff
@@ -621,6 +663,22 @@ impl<T: Config> Pallet<T> {
 			}
 		}
 
+		if !outgoing.is_empty() {
+			// Filter offboarded parachains from the upcoming upgrades list.
+			//
+			// We do it after the offboarding to get away with only a single read/write.
+			<Self as Store>::UpcomingUpgrades::mutate(|upcoming_upgrades| {
```

Comment:
Should `UpgradeCooldowns` be pruned as well?

Reply:
I see that it will be naturally pruned based on block, so it may be redundant.

Reply:
Yeah, similar to `UpcomingUpgrades`. However, we perform insertions based on `UpcomingUpgrades` but for `UpgradeCooldowns` we only perform removals.

That said, I am ok to add filtering for cooldowns.

Reply:
https://github.com/paritytech/polkadot/pull/3371/commits/d0c134e794682405a6374e0e293d66671e38803a